decidim-core/app/commands/decidim/create_report.rb

```diff
@@ -70,6 +71,19 @@ def send_report_notification_to_moderators
       end
     end
 
+    def send_report_notification_to_author
+      data = {
+        event: "decidim.events.reports.report_created",
+        event_class: Decidim::ReportCreatedEvent,
+        resource: @report.moderation.reportable,
+        extra: {
+          report_reason: @report.reason
+        },
+        affected_users: @report.moderation.reportable.try(:authors) || [@report.moderation.reportable.try(:author)]
+      }
+      Decidim::EventsManager.publish(data)
```

Comment:
I may lack of context but I feel like building this data structure should be in the `Decidim::ReportCreatedEvent` class, right? 🤔 
Something like:

```rb
event = Decidim::ReportCreatedEvent.new(@report)
Decidim::EventsManager.publish(event.to_data)
```

Reply:
That'd make a lot of sense. Still, all the events I've checked out are passed to the `EventManager` this way (the caller builds the event data), so I'm not sure we want to introduce a new way of doing it here.
What are your thoughts on this @mrcasals?

Reply:
You are... right. This makes more sense, but the current approach is the one Andrea used...